From 6e6ea465857b00bdc483e209922ad278dd21beec Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 21 Nov 2022 20:07:59 +0200
Subject: [PATCH] [SW-105079] gaudi3: pldm: skip hash verification for preboot

hash verification on pldm for preboot image loaded on SRAM takes
long time and slow downs the boot process. This patch add changes to
skip hash verification for preboot on pldm platform.

Change-Id: I6b045b710f5536caeeecd805a01975561fe92f05
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index a0ec387..e7afce1 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -584,7 +584,7 @@
 	return num_crc;
 }
 
-bool ppboot_is_valid_img_desc(struct boot_desc *desc)
+__weak bool hl_dev_is_valid_img_desc(struct boot_desc *desc)
 {
 	u32 addr, actual_size;
 	unsigned int num_crc;
@@ -603,6 +603,11 @@
 	return !ret;
 }
 
+bool ppboot_is_valid_img_desc(struct boot_desc *desc)
+{
+	return hl_dev_is_valid_img_desc(desc);
+}
+
 int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size)
 {
 	return boot_info_get_no_merge(type, data, max_size, copy_size);
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index 814a1ff..a42ddbe 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -11,6 +11,7 @@
 #include <uart.h>
 #include <pll.h>
 #include <arcboot_export_dev.h>
+#include <boot_desc.h>
 #include "hl.h"
 
 #define SYS_CLOCK_FREQUENCY	250000000ULL /* PCI PLL DIV3 SIMULATION CLK */
@@ -180,3 +181,8 @@
 {
 	security_rr_open_pcie_config();
 }
+
+bool hl_dev_is_valid_img_desc(struct boot_desc *desc)
+{
+	return true;
+}
